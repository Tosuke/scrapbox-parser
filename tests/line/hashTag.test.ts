/* global describe it expect */

describe('hashTag', () => {
  it('Simple hashTag', () => {
    expect('#tag').toMatchSnapshotWhenParsing()
  })

  it('Only `#` is not hashTag', () => {
    expect('#').toMatchSnapshotWhenParsing()
  })

  it('HashTag includes `#`', () => {
    expect('#hash#Tag').toMatchSnapshotWhenParsing()
  })

  it('HashTag in sentence with spaces', () => {
    expect('This is a #tag .').toMatchSnapshotWhenParsing()
  })

  it('HashTag in sentence without spaces is not hashTag', () => {
    expect('→#notTag←').toMatchSnapshotWhenParsing()
  })

  it('Multiple hashTag', () => {
    expect('#hoge #fuga #piyo').toMatchSnapshotWhenParsing()
  })
})
