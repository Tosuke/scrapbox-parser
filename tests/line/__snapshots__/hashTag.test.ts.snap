// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hashTag HashTag in sentence with spaces: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "text": "This is a ",
        "type": "plain",
      },
      Object {
        "href": "tag",
        "type": "hashTag",
      },
      Object {
        "text": " .",
        "type": "plain",
      },
    ],
    "type": "line",
  },
]
`;

exports[`hashTag HashTag in sentence without spaces is not hashTag: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "text": "→#notTag←",
        "type": "plain",
      },
    ],
    "type": "line",
  },
]
`;

exports[`hashTag HashTag includes \`#\`: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "href": "hash#Tag",
        "type": "hashTag",
      },
    ],
    "type": "line",
  },
]
`;

exports[`hashTag Multiple hashTag: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "href": "hoge",
        "type": "hashTag",
      },
      Object {
        "text": " ",
        "type": "plain",
      },
      Object {
        "href": "fuga",
        "type": "hashTag",
      },
      Object {
        "text": " ",
        "type": "plain",
      },
      Object {
        "href": "piyo",
        "type": "hashTag",
      },
    ],
    "type": "line",
  },
]
`;

exports[`hashTag Only \`#\` is not hashTag: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "text": "#",
        "type": "plain",
      },
    ],
    "type": "line",
  },
]
`;

exports[`hashTag Simple hashTag: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "href": "tag",
        "type": "hashTag",
      },
    ],
    "type": "line",
  },
]
`;
